### DISCUSSION TITLE
How to make a DXE driver module for edk2 and run on qemu

### DISCUSSION THREAD

**Discussion by shukladushyanta** [tianocore/edk2#10574] [General]
Hi, 

I want to add a dxe driver to my Edk2 code and the run it on qemu. I am using this article for the code and inf file. 

https://dev.to/machinehunter/how-to-build-dxe-driver-by-edk2-11h8

Here it shows to build for MdeModulePkg but on some places it show to build for OVMFPkg. I am new to all this and very confused. 

Can you tell me the process to make a DXE driver, where to add the source code(path and directory) modifications to all .dec and .dsc files. Also how to generate a build to launch the model on qemu. I mean which files to download. 

I am very confused. Please help!

**Comment by makubacki** [comment]
In case it is still helpful, check out the contents at https://github.com/tianocore-training/Tianocore_Training_Contents/wiki. That has presentations that explain the various files and how to use the open source platforms.
